### DISCUSSION TITLE
Live Reload - Infinite Loop

### DISCUSSION THREAD

**Discussion by i-am-the-arm** [remix-run/remix#4215] [Q&A] [answered]
Not sure if this is a bug or a bad configuration but I just started seeing an infinite loop when saving my project using Live Reload. Seems to be sporadic as to what causes it but I haven't been able to narrow down or consistently reproduce. See attached video for example. This is the result of me loading the page (initial load is fine), then clicking cmd+s to save the project, the browser automatically starts refreshing as the console loops as well. Any suggestions how I can start troubleshooting this issue?

Remix v1.7

This happens with all browsers and all terminal windows. 

edit: I just noticed this seems to happen when the browser window is focused. If I have the browser tab on the same page and save, it loops but if the browser tab is in a different scene (or whatever they're called on a mac), it doesn't. However, if I save then swipe over to that window, it loops. 

https://user-images.githubusercontent.com/22651504/190715515-b1682305-776c-48da-b57d-c9e9e1f5224a.mov

**Comment by kiliman** [comment]
That's odd as the web socket server should only close when you shutdown `remix dev` command.

Can you provide more details of your project? How are you running dev?

**Reply by i-am-the-arm** [reply]
Sure, what details would be helpful? I'm using `pnpm` and am ultimately running `remix dev`. Here are some relevant dependencies I'm using:

```
"dependencies": {
	  "react": "^18.2.0",
	  "react-dom": "^18.2.0",
	  "@remix-run/node": "^1.6.8",
	  "@remix-run/react": "^1.6.8",
	  "@remix-run/serve": "^1.6.8",
	  "@remix-run/server-runtime": "^1.6.8",
	    "remix-utils": "^3.3.0",
...

"devDependencies": {
    "@remix-run/dev": "^1.6.8",
    "@remix-run/eslint-config": "^1.6.8",
...
}
```

**Reply by kiliman** [reply]
Hmm... I've heard people sometimes have problems with `pnpm`. Can you try deleting `node_modules` and installing with `npm install`? Then see if you're still having the same problem.

**Reply by i-am-the-arm** [reply]
Looks like the same problem with `npm`...

**Reply by kiliman** [reply]
Then I don't know what to say. I don't have the issue here. Looks like something is killing the socket server and triggering it to restart. Good luck!

**Reply by i-am-the-arm** [reply]
Yeah, definitely a weird situation. Could something in the code cause it to break or is it more likely environmental? Best guess is fine, just looking for a good way to start deep diving :)

**Comment by i-am-the-arm** [comment, marked as answer]
This seemed to solve my issue for the time being. Looking for a more permanent solution but I'll continue digging into this. 

https://github.com/remix-run/remix/issues/1601#issuecomment-1155271096
